### PR TITLE
refactor aero_model_gasaerexch

### PR DESCRIPTION
**Pull request by tangshuaiqi**
This is the refactoring of gas-aerosol exchange.

### CODE REVIEW

**Review by jeff-cohere — APPROVED** [review]

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 1]
```diff
@@ -1330,87 +1330,88 @@ end subroutine aero_model_wetdep
...
-    real(r8) :: del_h2so4_aeruptk(ncol,pver)
+    real(r8), pointer :: dgnum(:,:,:)           ! aerosol diameter [m]
+    real(r8), pointer :: dgnumwet(:,:,:)        ! aerosol wet diameter [m]
+    real(r8), pointer :: wetdens(:,:,:)         ! aerosol wet density [kg/m3] 
+    real(r8), pointer :: pblh(:)                ! pbl height [m]
 
-    real(r8), pointer :: dgnum(:,:,:), dgnumwet(:,:,:), wetdens(:,:,:)
-    real(r8), pointer :: pblh(:)                    ! pbl height (m)
-    real(r8)  :: fldcw_all(ncol,pver,gas_pcnst)     ! all gas mass mixing ratio [kg/kg]
+    real(r8), dimension(ncol) :: dvmrdt_col     ! column-integrated tendency from chemistry [kg/kg/m2/s]
```
Comment (brigaudet) [inline components/eam/src/chemistry/modal_aero/aero_model.F90]: Wouldn't the units for this be [kg/(m2-s)]?  (or more specifically, [kg (mol/mol) / (m2-s)]?

**Review by singhbalwinder — APPROVED** [review]
Looks good! Please look into @brigaudet's question about the units.

**Review by brigaudet — APPROVED** [review]
Looks fine to me, just a units question.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 1 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/aero_model.F90]: Yes, I corrected it with kg/m2/s

**Comment by tangshuaiqi** [comment]
All tests passed. merge into refactor-maint-2.0

  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**